lib/pxelinux_cfg.py

```diff
@@ -93,14 +108,21 @@ def get_macfile(node):
 def get_boot_record_for_os(node, os_id):
 
     if os_id == id_local_boot:
-        cfg = pxe_common_settings + pxe_local_boot
+        cfg = pxe_local_boot
+    elif os_id == id_maintenance_boot:
+        cfg = pxe_main_menu.format(
+                             maintenance_image_kernel,
+                             maintenance_image_initrd,
+                             default_pxe_server)
     else:
         image = re.compile('^oem-').sub('', os_id)
         image = re.compile('-\d+\.\d+\.\d+').sub('', image)
-        cfg = pxe_common_os_boot_tmpl.format(os_id, os_id,
-                                             default_pxe_server, os_id, image)
+        cfg = pxe_common_os_boot_tmpl.format(
+                            os_id, os_id,
+                            default_pxe_server, os_id, image)
```

Comment:
This is kind of magic using same argument as os_id, can this format has a named arguments for replacement?

Reply:
done